packages/website/src/components/ast/tsUtils.ts

```diff
@@ -0,0 +1,83 @@
+interface TsParsedEnums {
+  SyntaxKind: Record<number, string>;
+  NodeFlags: Record<number, string>;
+  TokenFlags: Record<number, string>;
+  ModifierFlags: Record<number, string>;
+  ObjectFlags: Record<number, string>;
+  SymbolFlags: Record<number, string>;
+  FlowFlags: Record<number, string>;
+  TypeFlags: Record<number, string>;
+  ScriptKind: Record<number, string>;
+  TransformFlags: Record<number, string>;
+  ScriptTarget: Record<number, string>;
+  LanguageVariant: Record<number, string>;
+}
+
+export function extractEnum(
+  obj: Record<number, string | number>,
+): Record<number, string> {
+  const result: Record<number, string> = {};
+  const keys = Object.entries(obj);
+  for (const [name, value] of keys) {
+    if (typeof value === 'number') {
+      if (!(value in result)) {
+        result[value] = name;
+      }
+    }
+  }
+  return result;
+}
+
+declare global {
+  interface Window {
+    tsEnum: TsParsedEnums;
+  }
+}
+
+export function getTsEnum(type: keyof TsParsedEnums): Record<number, string> {
+  if (!window.tsEnum) {
+    window.tsEnum = {
```

Comment:
[Nitpick]

```suggestion
  window.tsEnum ??= {
```

Reply:
i actually don't like this being exposed in window, i'm going to change that